src/transformers/trainer.py

```diff
@@ -341,8 +341,8 @@ def _setup_wandb(self):
                 'Automatic Weights & Biases logging enabled, to disable set os.environ["WANDB_DISABLED"] = "true"'
             )
             wandb.init(project=os.getenv("WANDB_PROJECT", "huggingface"), config=vars(self.args))
-            # keep track of model topology and gradients
-            if os.getenv("WANDB_WATCH") != "false":
+            # keep track of model topology and gradients, unsupported on TPU
+            if not is_torch_tpu_available() and os.getenv("WANDB_WATCH") != "false":
```

Comment:
We also noticed some issues with multiprocessing that are currently being investigated.
In the meantime could you also disable watching when `self.args.n_gpu > 1`

Reply:
I have runs that utilize multiple GPUs which have gradients logged into Wandb. Maybe you are facing some other issue.

Reply:
Did you observe any reduced speed? When you look at your GPU usage on your runs do some of them seem underutilized?

It seems to be a common problems with pytorch hooks and distributed computing: https://discuss.pytorch.org/t/analyzing-hook-drastically-decreases-performance/53922/3

Reply:
Not really, the difference in utilization between GPUs is not too much. Though GPU 0 consumes more memory, maybe because it does aggregation operations.